src/hepattn/models/decoder.py

```diff
@@ -251,8 +272,10 @@ def forward(
         # Update key/constituent embeddings with the query/object embeddings
         if self.bidirectional_ca:
             if attn_mask is not None:
+                if self.attn_kwargs.get("attn_type", "torch") == "flex":
+                    assert attn_mask_transpose is not None, "attn_mask_transpose must be provided for flex attention"
```

Comment:
See above, nicer to store `self.attn_type` from the `kwargs` initially and then use it here